compiler/main.nim

```diff
@@ -299,18 +304,21 @@ proc mainCommand*(graph: ModuleGraph) =
     else:
       loadConfigs(DocConfig, cache, conf, graph.idgen)
       commandRst2Html(cache, conf)
-  of cmdRst2tex:
+  of cmdRst2tex, cmdDoc2tex:
```

Comment:
my intuition is that `cmdDoc2tex` should be more similar to `cmdDoc2` and hence should be grouped with it instead, eg so that the other statements are taken into account; wdyt?

Reply:
I had the same intuition initially. In practice it was a bit easier to insert it into `cmdRst2tex` section.
And there is no really significant impact, it can be written both ways.

Reply:
doc2tex generates a ton of `LockLevel` warnings for eg, which the code path for `cmdDoc2` handles; it makes sense: the frontend for  `cmdDoc2` and  `cmdDoc2tex` is the same.

this block is not relevant for doc2tex but is executed:
```
    for warn in [warnRedefinitionOfLabel, warnUnknownSubstitutionX,
                 warnLanguageXNotSupported,
                 warnFieldXNotSupported, warnRstStyle]:
      conf.setNoteDefaults(warn, true)
```

this block is relevant for doc2tex but is not executed:
```
      conf.setNoteDefaults(warnLockLevel, false) # issue #13218
      conf.setNoteDefaults(warnRedefinitionOfLabel, false) # issue #13218
        # because currently generates lots of false positives due to conflation
        # of labels links in doc comments, e.g. for random.rand:
        #  ## * `rand proc<#rand,Rand,Natural>`_ that returns an integer
        #  ## * `rand proc<#rand,Rand,range[]>`_ that returns a float
```


see what i mean? ;-)

a viable alternative is to use an auxiliary template/proc to call the shared part

Reply:
> this block is not relevant for doc2tex but is executed:

why is it not relevant for `doc2tex`? May be it is the other way around, it should be enabled for `cmdDoc2` also? We need RST/Markdown warning for `nim` files also, am I wrong?..

Reply:
@a-mr just saw your response, yes, it looks like it should be relevant for both cmdDoc2 and cmdDoc2tex

(but `this block is relevant for doc2tex but is not executed:` is still relevant)